File: app/src/main/java/com/arkapp/partyplanner/data/room/SummaryDetailsDao.java
package com.arkapp.partyplanner.data.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.arkapp.partyplanner.data.models.SummaryDetails;

import java.util.List;

/**
 * Created by Abdul Rehman on 17-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */


/**
 * This class is used to define query for the Summary table
 */

@Dao
public interface SummaryDetailsDao {
    @Insert(onConflict = 1)
    void insert(SummaryDetails summaryDetails);

    @Query("SELECT * FROM SUMMARY WHERE uid = :userUid")
    List<SummaryDetails> getUserSummary(int userUid);

    @Query("DELETE FROM SUMMARY WHERE uid = :userUid")
    void delete(int userUid);
}

File: app/src/main/java/com/arkapp/partyplanner/data/room/SummaryDetailsDao.kt
package com.arkapp.partyplanner.data.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.arkapp.partyplanner.data.models.SummaryDetails

/**
 * Created by Abdul Rehman on 17-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */


/**
 * This class is used to define query for the Summary table
 * */

@Dao
interface SummaryDetailsDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(vararg summaryDetails: SummaryDetails)

    @Query("SELECT * FROM SUMMARY WHERE uid = :userUid")
    suspend fun getUserSummary(userUid: Int): List<SummaryDetails>

    @Query("DELETE FROM SUMMARY WHERE uid = :userUid")
    suspend fun delete(userUid: Int)
}